src/data_structures/graph.js

```diff
@@ -53,4 +53,21 @@ Graph.prototype.edge = function(a, b) {
   return this.adjList[_(a)][_(b)];
 };
 
+Graph.prototype.reverse = function() {
+  var self = this;
+  var r = new Graph(this.directed);
+
+  self.vertices.forEach(function(v) {
+    r.addVertex(v);
+  });
+
+  Object.keys(self.adjList).forEach(function(a) {
+    Object.keys(self.adjList[a]).forEach(function(b) {
+      r.adjList[b][a] = self.adjList[a][b];
+    });
+  });
+
```

Comment:
This is fine and works, but maybe it would be better if you used the methods that already do those things to ignore the aspect that we use an adjacency list here, like this:
```js
self.vertices.forEach(function(a) {
  self.neighbors(a).forEach(function(b) {
    self.addEdge(b, a, self.edge(a, b));
  }
}
```

Because if we change how we add edges or vertices, we can still change in a single place that is reused.